src/scrabble.js

```diff
@@ -1,5 +1,191 @@
+        newWord[i].includes('J') ||
+        newWord[i].includes('X') ||
+        newWord[i].includes('j') ||
+        newWord[i].includes('x')
+      ) {
+        sum += 8
+      }
+      if (
+        newWord[i].includes('Q') ||
+        newWord[i].includes('Z') ||
+        newWord[i].includes('q') ||
+        newWord[i].includes('z')
+      ) {
+        sum += 10
+      }
+    }
+    return sum
+  }
+}
+
+function doubleLetter(aWord) {
+  const regex1 = /{[A-Z|a-z]}/g
+  const found1 = aWord.match(regex1)
+  const doubleLetter = normalScore(found1) * 2
+  return doubleLetter
+}
+
+console.log('Double letter test', doubleLetter('{A}{A}'))
+
+function trippleLetter(aWord) {
+  const regex2 = /\[[A-Z | a-z]]/g
+  const found2 = aWord.match(regex2)
+  const trippleLetter = normalScore(found2) * 3
+  return trippleLetter
+}
+
+function normalLetter(aWord) {
+  const split = aWord.replaceAll(/{[A-Z | a-z]}/g, '')
+  const split2 = split.replaceAll(/\[[A-Z | a-z]]/g, '')
 
+  console.log('split3', split2.toString())
+  console.log('First Normal Score', normalScore(split2))
+  return normalScore(split2)
 }
 
+function doubleWords(aWord5) {
+  const myLength = aWord5.length
+  if (aWord5[0] === '{' && aWord5[myLength - 1] === '}') {
+    const newWord = aWord5.slice(1, myLength - 1)
+    console.log('newWord', newWord)
+    const doubleWords = normalScore(newWord) * 2
+    return doubleWords
+  }
+}
+
+console.log('Double word score', doubleWords('{AAAAA}'))
+
+function scrabble(aWord) {
+  if (typeof aWord === 'string') {
+    const myLength = aWord.length
+    if (aWord[0] === '[' && aWord[myLength - 1] === ']') {
+      const newWord2 = aWord.slice(1, myLength - 1)
+      console.log('newWord', newWord2)
+      if (newWord2[0] === '{') {
+        const tripleWords =
+          (normalScore(newWord2) * 3 +
+            doubleLetter(newWord2) +
+            trippleLetter(newWord2)) *
+          2
+        return tripleWords
+      } else {
+        const tripleWords =
+          normalLetter(newWord2) * 3 +
+          doubleLetter(newWord2) +
+          trippleLetter(newWord2)
+        return tripleWords
+      }
+    }
+    const myLength2 = aWord.length
+    if (aWord[0] === '{' && aWord[myLength2 - 1] === '}' && aWord[2] !== '}') {
+      const newWord3 = aWord.slice(1, myLength - 1)
+      console.log('newWord here', newWord3)
+      console.log('normal score', normalScore(newWord3))
+      console.log('double letter', doubleLetter(newWord3))
+      console.log('triple letter', trippleLetter(newWord3))
+      const doubleWords =
+        (normalLetter(newWord3) +
+          doubleLetter(newWord3) +
+          trippleLetter(newWord3)) *
+        2
+      return doubleWords
+    }
+    if (
+      (aWord.includes('[') && aWord.includes(']')) ||
+      (aWord.includes('{') && aWord.includes('}'))
+    ) {
+      const finalScore =
+        normalLetter(aWord) + doubleLetter(aWord) + trippleLetter(aWord)
+
+      return finalScore
+    }
+    if (
```

Comment:
Also clever way to check that there is never only one [ in the string and that there is always two, and there is no invalid characters. However, if there are characters entered like ? $ % / £ €, or other characters that shouldn't be included I didn't see a way for the program to return 0.